azure-pipelines.yml

```diff
@@ -240,14 +243,17 @@ jobs:
       cmakeArgs: '..'
 
   - script: |
+      set -eo pipefail
       make clangformat
       git diff
       if [ "$(git diff $(Build.SourceVersion))" != "" ]; then exit 1; fi
 
     workingDirectory: build
+    failOnStderr: true
     displayName: 'Clang-Format'
 
   - script: |
+      set -eo pipefail
       clang-tidy-8  src/override/malloc.cc  -header-filter="`pwd`/*" -export-fixes=tidy.fail -- -std=c++17 -mcx16
       if [ -f tidy.fail ] ; then exit 1 ; fi
```

Comment:
If you add `-warnings-as-errors='*'`, clang-tidy will report errors and exit with a non-zero code. You should then be able to remove this line, and the `-export-fixes`

Unrelated, but why do we have clang-tidy on just this file?

Reply:
We can try and expand it.